pkg/tekton/metapipeline/client.go

```diff
@@ -0,0 +1,24 @@
+package metapipeline
+
+import (
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/tekton"
+)
+
+// Client defines the interface for meta pipeline creation and application.
+type Client interface {
+	// Create creates the Tekton CRDs needed for executing the pipeline as defined by the input parameters
+	Create(pullRef PullRef, pipelineType PipelineKind, context string, envs map[string]string, labels map[string]string) (kube.PromoteStepActivityKey, tekton.CRDWrapper, error)
```

Comment:
I would extract the list of arguments into an `options` struct. This will allow us to extend the arguments easier in future.